src/ergw_aaa_radius.erl

```diff
@@ -171,10 +171,21 @@ validate_ip(_Opt, {_,_,_,_,_,_,_,_} = IP) ->
 validate_ip(Opt, IP) ->
     throw({error, {options, {Opt, IP}}}).
 
+validate_server_spec_list(_Opt, [], Acc) ->
+    lists:reverse(Acc);
+    
+validate_server_spec_list(Opt, [{NAS_Type, {IP, Port, Secret}} = H | T], Acc) 
+    when is_integer(Port), is_binary(Secret),
+        (NAS_Type =:= default orelse NAS_Type =:= backup) ->
+    validate_ip(Opt, IP),
+    validate_server_spec_list(Opt, T, [H | Acc]).
+    
 validate_server_spec(Opt, {IP, Port, Secret} = Value)
   when is_integer(Port), is_binary(Secret) ->
     validate_ip(Opt, IP),
     Value;
+validate_server_spec(Opt, Dests) when is_list(Dests) ->
+    validate_server_spec_list(Opt, Dests, []);
```

Comment:
you should turn the proplist into a map (there are plenty examples for this in the other config checkers). This could also be used to simplify the send_request method.

Reply:
done